File: theme-replace.py
import os

src_dir = os.path.join(os.path.dirname(__file__), 'src')

def replace_in_file(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    new_content = content
    # Theme replacements
    new_content = new_content.replace('indigo-', 'emerald-')
    new_content = new_content.replace('slate-', 'stone-')
    new_content = new_content.replace('purple-', 'amber-')
    new_content = new_content.replace('blue-', 'teal-')
    new_content = new_content.replace('bg-[#f3f5f9]', 'bg-stone-50')
    new_content = new_content.replace('bg-[#0a0a0e]', 'bg-[#1c1917]') # stone-900ish

    if content != new_content:
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Updated {file_path}")

def traverse_dir(directory):
    for root, _, files in os.walk(directory):
        for file in files:
            if file.endswith('.tsx') or file.endswith('.ts'):
                replace_in_file(os.path.join(root, file))

traverse_dir(src_dir)
print("Done.")

File: src/theme-replace.js
const fs = require('fs');
const path = require('path');

const srcDir = path.join(__dirname, 'src');

function replaceInFile(filePath) {
  const content = fs.readFileSync(filePath, 'utf-8');
  let newContent = content;
  
  // Color replacements
  newContent = newContent.replace(/indigo-/g, 'emerald-');
  newContent = newContent.replace(/slate-/g, 'stone-');
  newContent = newContent.replace(/purple-/g, 'amber-');
  newContent = newContent.replace(/blue-/g, 'teal-');
  newContent = newContent.replace(/bg-\[\#f3f5f9\]/g, 'bg-stone-50');
  newContent = newContent.replace(/bg-\[\#0a0a0e\]/g, 'bg-[#1c1917]'); // stone-900ish

  if (content !== newContent) {
    fs.writeFileSync(filePath, newContent, 'utf-8');
    console.log(`Updated ${filePath}`);
  }
}

function traverseDir(dir) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const fullPath = path.join(dir, file);
    if (fs.statSync(fullPath).isDirectory()) {
      traverseDir(fullPath);
    } else if (fullPath.endsWith('.tsx') || fullPath.endsWith('.ts')) {
      replaceInFile(fullPath);
    }
  }
}

traverseDir(srcDir);
console.log('Done.');
